src/mesido/esdl/asset_to_component_base.py

```diff
@@ -457,15 +458,25 @@ def _set_electricity_current_nominal_and_max(
     def _get_connected_q_max(self, asset: Asset) -> float:
         if asset.in_ports is not None and asset.asset_type != "Electrolyzer":
             for port in asset.in_ports:
+                convert_density_units = 1.0
+                energy_reference_j_kg = 1.0
+                if isinstance(port.carrier, esdl.HeatCommodity):
+                    convert_density_units = 1.0e3  # convert g/m3 to kg/m3 if needed
+                    energy_reference_j_kg = get_internal_energy(asset.name, port.carrier)
+                else:
+                    # heat_value / rho * minimum_dT => [J/m3K] / [kg/m3] * 1.0 [K] => [J/kg]
+                    energy_reference_j_kg = HEAT_STORAGE_M3_WATER_PER_DEGREE_CELCIUS / 988.0
```

Comment:
Shouldn't the energy_reference_j_kg be switched around? In case of the heat commodity you want to use the cp (HEAT_STORAGE_M3_WATER_PER_DEGREE_CELSIUS).
To fix it, change "if isinstance" to "if not isinstance"

Reply:
My mistake. It has been fixed.